lib/chat_models/chat_vertex_ai.ex

```diff
@@ -482,9 +482,13 @@ defmodule LangChain.ChatModels.ChatVertexAI do
 
   def do_process_response(response, message_type \\ Message)
 
-  def do_process_response(%{"candidates" => candidates}, message_type) when is_list(candidates) do
+  def do_process_response(%{"candidates" => candidates} = data, message_type)
+      when is_list(candidates) do
+    token_usage = get_token_usage(data)
+
```

Comment:
After determining that we have token usage, it should fire a callback with the information:

```elixir
Callbacks.fire(model.callbacks, :on_llm_token_usage, [token_usage])
```

You can find numerous examples in the code when you search for `:on_llm_token_usage`.

I'd love to see test coverage of firing the event.